framework/src/modules/chain/bft/errors.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { FrameworkError } = require('../../../errors');
+
+class BFTError extends FrameworkError {}
```

Comment:
what is the point of extending `FrameworkError`? Prefer to just extend `Error`

Reply:
We have a base error constructor and every error in framework should be extended by it. There is no harm in it and could be useful in future.

Reply:
@yatki I would do it on your request 😄

Reply:
While refactoring, I noticed that its very tedious to match against long string messages to see if that particular error occurred. So its better to have different error constructors for each error type for now. 

I agree to the approach that we must have less error constructors, but for that we need to introduce error codes in these system to pinpoint the exact error quickly rather matching long string messages. That usually end up in doing regular expression matching on the error messages.

Reply:
If we have a separate issue to follow up on this, we can ignore it for now. 👍  @nazarhussain could you please create the issue, and reference to this discussion?

Reply:
@yatki Created a separate issue to refine that aspect fo the SDK https://github.com/LiskHQ/lisk-sdk/issues/3899